pkg/domain/randomnumbergenerator.go

```diff
@@ -0,0 +1,10 @@
+package domain
+
+import (
+	"github.com/google/uuid"
+)
+
+// RandomNumberGenerator generates a sufficiently random UUID
+type RandomNumberGenerator interface {
+	NewRandom() (uuid.UUID, error)
```

Comment:
suggestion: All current usage of this output is converting it into a `string` type and not actually using any of the other implementation details of the `uuid.UUID` type. I believe it would simplify both the usage and the tests if this returned a string instead of the 3rd party type.